src/app/organisms/emoji-board/recent.js

```diff
@@ -0,0 +1,31 @@
+import initMatrix from '../../../client/initMatrix';
+import { emojis } from './emoji';
+
+const eventType = 'io.element.recent_emoji';
+
+function get() {
+  return initMatrix.matrixClient.getAccountData(eventType).getContent().recent_emoji ?? [];
+}
+
+export function getRecentEmojis(limit) {
+  return get()
+    .sort((a, b) => b[1] - a[1])
+    .slice(0, limit)
+    .map(([unicode]) => emojis.find((e) => e.unicode === unicode));
+}
+
+export function addRecentEmoji(unicode) {
+  const recent = get();
+  const i = recent.findIndex(([u]) => u === unicode);
+  let entry;
+  if (i < 0) {
+    entry = [unicode, 1];
+  } else {
+    [entry] = recent.splice(i, 1);
+    entry[1] += 1;
+  }
+  recent.unshift(entry);
```

Comment:
using `recent.push` if we are adding new emoji and directly updating the `recent[i][1] += 1;` will be faster approach.

Reply:
Element web expects the emojis to be sorted by last used. That's we we do this dance of removing it from the array and adding it back on

Reply:
that's ok